lib/developer_portal/app/controllers/developer_portal/base_controller.rb

```diff
@@ -13,6 +13,14 @@ class DeveloperPortal::BaseController < DeveloperPortal::ApplicationController
 
   before_action :verify_requested_format!
 
+  content_security_policy do |policy|
+    policy.default_src '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+    policy.font_src    '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+    policy.img_src     '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+    policy.script_src  '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+    policy.style_src   '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+  end
```

Comment:
how did you come up with these values?

Reply:
I wanted to allow everything for the dev portal, I added all sources https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources

Reply:
It increases the scope of the PR but it appears to me that allowing everything is not an improvement compared to not setting the header. Do you think it will be possible to have it user configurable and we default to safer defaults?

Reply:
hmm, seems not possible OOB. There is a project but I don't know how reasonable that is 
https://github.com/DocSpring/better_content_security_policy

Reply:
> allowing everything is not an improvement compared to not setting the header

We only allow everything for the developer portal, we are much more restrictive in the admin portal.

About the dev. portal, agreed is not much advantage, at least I expect Rapidast to not raise a warning anymore, since the CSP header is present. Now we are explicitly stating that we allow everything, compared to before when we allowed everything because nobody worked on this.

> There is a project but I don't know how reasonable that is
https://github.com/DocSpring/better_content_security_policy

I don't think that gem is useful because Rails already includes a builtin functionality to set a different CSP at controller level. It's in fact what we use to setup a different CSP for the dev. portal. Allowing users to setup their own CSP is possible, I would say, but would require a lot of work, including UI work. Out of this PR scope for sure.

Reply:
It's not the point to just make the warning disappear. We need to decide whether it makes sense to allow individual setting or not. 

This gem's purpose as far as I see is to make the built-in rails mechanism more flexible. As it is rather static at this time. You can't override per cal it appears. We may also add a controller before hook or a middleware to add the headers depending on some user settings for the dev portals.

There seem to be also different options, for example https://content-security-policy.com/strict-dynamic/

I'm fine thought that this PR is fine not to address this. Just we need to decide whether this would be a final fix or a step.

Reply:
Checking the gem `README` I don't see how's the gem better than the builtin rails method. It seems to me they do exactly the same.

> Just we need to decide whether this would be a final fix or a step.

I'm fine with both. But yeah, better leave the discussion for another Jira/PR

Reply:
Can we mock CSP response only for that feature? Or list the test hosts as a list instead of just `*`?

Reply:
I don't know how to mock CSP headers for a scenario, but I suspect is going to be (cu)cumbersome. About listing the asset hosts, we can just allow `cdn.3scale.localhost:*` and it will work for all scenarios. I updated it.

Reply:
I was looking for an option to avoid test specific settings so that when running the tests, we know that CSP is same as in production.

https://github.com/3scale/porta/pull/3833/commits/f0ed4c85aa05d7ae8f690eff3e40677208e165ca#diff-8e6d893db354f85583e669d561d33c1838b19f805bc76fa7525efc5410404c90R12

Reply:
That link is outdated. This is how it looks now: https://github.com/3scale/porta/pull/3833/files#diff-8e6d893db354f85583e669d561d33c1838b19f805bc76fa7525efc5410404c90R11-R20

Reply:
That look much better!

 Can we just abstract this into a method that both - prod and test use instead of having the code in two places and potentially getting out of sync?
 
 Also why do we need host `*`? don't we know the expected test asset_hosts hostnames in advance so that we only allow these? So that we can notice during testing whether we access resources from unexpected hosts (e.g. external hosts).

Reply:
> Can we just abstract this into a method that both - prod and test use instead of having the code in two places and potentially getting out of sync?

Done at https://github.com/3scale/porta/pull/3833/commits/7650f4507b4b18cb148b409581d5c40041c1006d

> Also why do we need host *?

In the test suite do you mean? I removed the wildcard from the domain name, now it's always `cdn.3scale.localhost:*`. See https://github.com/3scale/porta/pull/3833/files#diff-8e6d893db354f85583e669d561d33c1838b19f805bc76fa7525efc5410404c90R11